cabal-install/Distribution/Client/FileMonitor.hs

```diff
@@ -793,19 +792,19 @@ buildMonitorStateFileSet :: Maybe MonitorTimestamp -- ^ optional: timestamp
                                               --   relative to root
                          -> IO MonitorStateFileSet
 buildMonitorStateFileSet mstartTime hashcache root =
-    go Map.empty []
+    go [] []
   where
-    go :: Map FilePath MonitorStateFile -> [MonitorStateGlob]
+    go :: [MonitorStateFile] -> [MonitorStateGlob]
        -> [MonitorFilePath] -> IO MonitorStateFileSet
     go !singlePaths !globPaths [] =
-      return (MonitorStateFileSet singlePaths globPaths)
+      return (MonitorStateFileSet (reverse singlePaths) (reverse globPaths))
```

Comment:
This is so the internal [MonitorStateFile] is stored in the same order as the original monitor specs (since `go` accumulates the lists in reverse order of course). Not a big deal but makes the unit tests a bit more sane. Also, since we extract the set of file monitors to include into other monitors, if we have to debug something in future and it'd be a bit confusing if one also have to deal with the order flipping.